rescript/get_unite.py

```diff
@@ -0,0 +1,159 @@
+
+def _unite_get_tgz(
+    url: str = None, download_path: str = None, retries: int = 10
+) -> str:
+    """Download compressed database"""
+    for retry in range(retries):
+        # Track downloaded size
+        file_size = 0
+        try:
+            response = requests.get(url, stream=True)
+            # Save .tgz file
+            unite_file_path = os.path.join(download_path, "unitefile.tar.gz")
+            with open(unite_file_path, "wb") as f:
+                for chunk in response.iter_content(chunk_size=8192):
+                    if chunk:
+                        f.write(chunk)
+                        file_size += len(chunk)
+            # Check if the downloaded size matches the expected size
+            if file_size == int(response.headers.get("content-length", 0)):
+                return unite_file_path  # done!
+            else:
+                raise ValueError("File download incomplete")
+        except HTTPError as e:
+            print(
+                "Request failed with code "
+                + str(e.response.status_code)
+                + ", on try "
+                + str(retry)
+            )
+        except ValueError:
+            print("File incomplete, on try " + str(retry))
+
+
+def _unite_get_artifacts(
+    tgz_file: str = None, cluster_id: str = None
+) -> (DataFrame, DNAIterator):
+    """
+    Find and import files with matching cluster_id from .tgz
+
+    Returns: Tuple containing tax_results and seq_results
+    """
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # Extract from the .tgz file
+        with tarfile.open(tgz_file, "r:gz") as tar:
+            # Keep only _dev files
+            members = [m for m in tar.getmembers() if "_dev" in m.name]
+            if not members:
+                raise ValueError("No '_dev' files found")
+            for member in members:
+                # Keep only base name
+                member.name = os.path.basename(member.name)
+                tar.extract(member, path=tmpdirname)
+        # Find and import the raw files...
+        for root, dirs, files in os.walk(tmpdirname):
+            # ... with the matching cluster_id
+            filtered_files = [
+                f for f in files if f.split("_")[4] == cluster_id
+            ]
+            if not filtered_files or len(filtered_files) > 2:
+                raise ValueError(
+                    "Found "
+                    + str(len(filtered_files))
+                    + " files found with cluster_id = "
+                    + cluster_id
+                )
+            for file in filtered_files:
+                fp = os.path.join(root, file)
+                if file.endswith(".txt"):
+                    taxa = TaxonomyFormat(fp, mode="r").view(DataFrame)
+                elif file.endswith(".fasta"):
+                    seqs = DNAFASTAFormat(fp, mode="r").view(DNAIterator)
+    return taxa, seqs
+
+
+def get_unite_data(
+    version: str = None,
+    taxon_group: str = None,
```

Comment:
Explicitly, set defaults to the latest version here, *e.g.* [gtdb](https://github.com/bokulich-lab/RESCRIPt/blob/946728467532f066c8bb750c3d8d1b8af02773b9/rescript/get_gtdb.py#L29). Otherwise this will not work as you intend, even though you set the settings as optional:

```
qiime rescript get-unite-data --output-dir test-unite-dl --verbose
```

This command returns:
```
Plugin error from rescript:
  None
See above for debug info.
```